tests/common/helpers/pfc_storm.py

```diff
@@ -0,0 +1,258 @@
+                pfc_frames_number(int) : Number of PFC frames to generate. default: 100000
+                pfc_gen_file(string): Script which generates the PFC traffic. default: 'pfc_gen.py'
+                Other keys: 'pfc_storm_defer_time', 'pfc_storm_stop_defer_time', 'pfc_asym'
+        """
+        self.dut = duthost
+        dut_facts = duthost.setup()['ansible_facts']
+        hostvars = self.dut.host.options['variable_manager']._hostvars[self.dut.hostname]
+        self.inventory = hostvars['inventory_file'].split('/')[-1]
+        self.ip_addr = dut_facts['ansible_eth0']['ipv4']['address']
+        self.fanout_info = fanout_graph_facts
+        self.fanout_hosts = fanouthosts
+        self.pfc_gen_file = kwargs.pop('pfc_gen_file', "pfc_gen.py")
+        self.pfc_queue_idx = kwargs.pop('pfc_queue_index', 3)
+        self.pfc_frames_number = kwargs.pop('pfc_frames_number', 100000)
+        self.peer_info = kwargs.pop('peer_info')
+        self._validate_params(expected_args=['pfc_fanout_interface', 'peerdevice'])
+        if 'hwsku' not in self.peer_info:
+            self._populate_peer_hwsku()
+        self.platform_name = None
+        self.update_platform_name()
+        self._populate_optional_params(kwargs)
+        self.peer_device = self.fanout_hosts[self.peer_info['peerdevice']]
+
+    def _populate_peer_hwsku(self):
+        """
+        Find out the hwsku associated with the fanout
+        """
+        peer_dev_info = self.fanout_info[self.peer_info['peerdevice']]['device_info']
+        self.peer_info['hwsku'] = peer_dev_info['HwSku']
+
+    def _validate_params(self, **params):
+        """
+        Validate if all the needed keys are present
+        """
+        expected_args = params.get('expected_args')
+        peer_info_keys = self.peer_info.keys()
+        if not all(elem in peer_info_keys for elem in expected_args):
+            raise MissingInputError("Peer_info does not contain all the keys,"
+                                    "Expected args: {}".format(expected_args))
+
+    def _populate_optional_params(self, kwargs):
+        """
+        Create var and assign values if any the following keys are present
+        'pfc_storm_defer_time', 'pfc_storm_stop_defer_time', 'pfc_asym'
+        """
+        if len(kwargs) > 0:
+            for key in kwargs:
+                setattr(self, key, kwargs.get(key))
+        kwargs.clear()
+
+    def _create_pfc_gen(self):
+        """
+        Create the pfc generation file on the fanout if it does not exist
+        """
+        out = self.peer_device.stat(path="/mnt/flash")
+        if not out['stat']['exists'] or not out['stat']['isdir']:
+            self.peer_device.file(path="/mnt/flash/{}".format(self.pfc_gen_file), state="touch")
+
+    def deploy_pfc_gen(self):
+        """
+        Deploy the pfc generation file on the fanout
+        """
+        if 'arista' in self.peer_info['hwsku'].lower():
+            self._create_pfc_gen()
```

Comment:
why do we need to create the file and then copy it? Would copy not suffice?

Reply:
good q. I saw that if i just copy the owner is root vs eosadmin if i touch the file and then copy. So left it as is.

Reply:
The [copy](https://docs.ansible.com/ansible/2.8/modules/copy_module.html#copy-module) module does have mode and group to manipulate file ownership

Reply:
will use that. thanks

Reply:
explored this a bit. The current code checks if the dir is present and if not, touch the file in the dir. so that will create the dir as well if it is nonexistent. 
if I just use copy alone, the task will fail if the dest dir is nonexistent.